build.sh

```diff
@@ -11,6 +31,42 @@ while [[ -h $source ]]; do
   # symlink file was located
   [[ $source != /* ]] && source="$scriptroot/$source"
 done
-
 scriptroot="$( cd -P "$( dirname "$source" )" && pwd )"
-"$scriptroot/eng/common/build.sh" --build --restore $@
+
+# Import Arcade functions
+. "$scriptroot/eng/common/tools.sh"
+
+function BuildBootstrapCompiler {
+  InitializeDotNetCli true
+  # Build linker
+  echo "Building illink"
+
+  local dir="$artifacts_dir/bootstrap"
```

Comment:
I'd prefer to do this in msbuild so that it works on all platforms. Ideally you'd be able to run `dotnet test test/Mono.Linker.Tests/Mono.Linker.Tests.csproj /p:BootstrapCsc=true` or similar and it would set up the bootstrap compiler.

Reply:
Per our previous conversation, I'd rather not do this in MSBuild as it would involve either creating another project file, which seems like overkill to me, or having a circular dependency where the linker is dependent upon itself.

This is expressible in MSBuild, but it doesn't seem like a particularly clear use of it.

Re: xplat -- happy to port this to our `cmd` files as well. Getting linux working was just the first and easiest step.

Reply:
I'd still prefer for this to work with `dotnet` tooling, but it's probably more important to get the extra validation working. And I can see that MSBuild might not be the clearest way to write it. :)